runtime/src/accounts_db.rs

```diff
@@ -3813,14 +3953,31 @@ impl AccountsDB {
         (result, sum)
     }
 
-    fn flatten_hashes(hashes: Vec<Vec<Hash>>) -> (Vec<Hash>, Measure, usize) {
-        // flatten vec/vec into 1d vec of hashes in order
+    fn flatten_hashes_and_hash(
+        hashes: Vec<Vec<Hash>>,
+        fanout: usize,
+        test: bool,
+    ) -> (Hash, Measure, usize) {
         let mut flat2_time = Measure::start("flat2");
-        let hashes: Vec<Hash> = hashes.into_iter().flatten().collect();
+
+        let offsets = CumulativeOffsets1D::from_raw(&hashes);
+        let hash_bkup = if test {
```

Comment:
I would have this path just in the test code itself. We should be able to call both paths and compare in a test.

Reply:
it was already also being tested in tests